tests/test_camera.py

```diff
@@ -0,0 +1,89 @@
+import os
+import unittest
+from unittest.mock import patch
+
+import cv2
+
+import sense.camera as camera
+
+cwd = os.getcwd()
+video_path = os.path.join(cwd, 'tests/resources/test_video.mp4')
+
+
+class TestVideoSource(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.video = camera.VideoSource(filename=video_path)
+
+    @patch('sense.camera.VideoSource.pad_to_square')
+    def test_get_image(self, mock_pad_to_square):
+        square_img_path = os.path.join(cwd, 'tests/resources/square_SENSE.png')
+        square_img = cv2.imread(square_img_path)
+        mock_pad_to_square.return_value = square_img
+        img, scaled_img = self.video.get_image()
+        self.assertTrue(mock_pad_to_square)
+        assert img.shape == scaled_img.shape
+
+    def test_pad_to_square(self):
+        img_path = os.path.join(cwd, 'tests/resources/SENSE.png')
+        img = cv2.imread(img_path)
+        max_length = max(img.shape[0:2])
+        new_img = self.video.pad_to_square(img)
+        assert new_img.shape == (max_length, max_length, 3)
+
+    def test_fps(self):
+        fps = self.video.get_fps()
+        assert fps == 12.0
+
+
+class TestVideoStream(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.video = camera.VideoSource(filename=video_path)
+        self.stream = camera.VideoStream(video_source=self.video, fps=12.0)
+
+    def test_stop(self):
+        self.stream.stop()
+        self.assertTrue(self.stream._shutdown)
+
+    def test_extract_image(self):
+        img_tuple = self.video.get_image()
+        self.stream.frames.put(img_tuple, False)
+        imgs = self.stream.get_image()
+        assert str(type(imgs[0])) == "<class 'numpy.ndarray'>"
```

Comment:
Why not directly check for the correct class?
```python
assert type(imgs[0]) == np.ndarray
```

Reply:
> Why not directly check for the correct class?
> 
> ```python
> assert type(imgs[0]) == np.ndarray
> ```

update